Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -44,16 +44,13 @@ open class ChatChannelListVC<ExtraData: UIExtraDataTypes>: UIViewController,
         _ collectionView: UICollectionView,
         cellForItemAt indexPath: IndexPath
     ) -> UICollectionViewCell {
-        let channel = controller.channels[indexPath.row]
-        
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
-        
-        if let channelView = cell.contentView.subviews.first as? ChatChannelView<ExtraData> {
-            channelView.channel = channel
-        } else {
-            let channelView = uiConfig.channelList.channelView.init(channel: channel, uiConfig: uiConfig)
-            cell.contentView.embed(channelView)
-        }
+        let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: "Cell",
+            for: indexPath
+        ) as! ChatChannelListCollectionViewCell<ExtraData>
+    
+        cell.uiConfig = uiConfig
```

Comment:
This needs to be done before. Should we do it in  `dequeueReusableCell` in our custom collection view subclass?

Reply:
Yes, we can wrap `cell.uiConfig = uiConfig` by overriding `dequeueReusableCell` in our custom collection-view but the issue with calling `init(config: UIConfig)` will still be opened.

Suggest to work on it in a dedicated task. 

It might happen we will occasionally come up with smth similar to SwiftUI's environment variables: we will be iterating up by view hierarchy checking if a superview provides a `config: UIConfig<ExtraData>` and use `.default` if it wasn't found.

Reply:
OK 👍